exporter-helper.php

```diff
@@ -15,139 +106,131 @@ function get_campaign_attachments( $post_ids ) {
 
 	global $wpdb;
 
-	// phpcs:disable
-	$sql = 'SELECT ID, guid, post_parent FROM %1$s WHERE post_type = \'attachment\'';
-	$prepared_sql = $wpdb->prepare(
-		$sql,
-		[
-			$wpdb->posts,
-		] );
-	$attachments = $wpdb->get_results( $prepared_sql, OBJECT_K );
-	// phpcs:enable
-
-	if ( empty( $attachments ) ) {
-		return $post_ids;
+	if ( empty( $post_ids ) ) {
+		return [];
 	}
 
-	$attachment_ids = [];
+	$sql = '
+		SELECT id
+		FROM %1$s
+		WHERE post_type = \'attachment\'
+		AND post_parent IN (' . generate_list_placeholders( $post_ids, 2 ) . ')';
 
-	$placeholders   = [];
-	$post_ids_count = count( $post_ids );
-	for ( $i = 2; $i < $post_ids_count + 2; $i++ ) {
-		$placeholders[] = "%$i\$d";
-	}
-	$placeholders = implode( ',', $placeholders );
+	$prepared_sql   = $wpdb->prepare(
+		$sql, // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+		array_merge( [ $wpdb->posts ], $post_ids )
+	);
+	$attachment_ids = $wpdb->get_results( $prepared_sql, OBJECT_K ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
 
 	/**
 	 * Post thumbnails
 	 */
-	if ( $post_ids ) {
-		$sql            = 'SELECT meta_value FROM %1$s  WHERE ( meta_key = \'_thumbnail_id\' or meta_key = \'background_image_id\' ) AND post_id IN(' . $placeholders . ')';
-		$values         = [];
-		$values[0]      = $wpdb->postmeta;
-		$values         = array_merge( $values, $post_ids );
-		$prepared_sql   = $wpdb->prepare( $sql, $values ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
-		$results        = $wpdb->get_results( $prepared_sql ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
-		$attachment_ids = [];
-		foreach ( (array) $results as $result ) {
-			$attachment_ids[] = $result->meta_value;
-		}
-	}
+	$sql = '
+		SELECT meta_value
+		FROM %1$s
```

Comment:
I'm not sure if there is actually much value in using a placeholder for the table name when making custom queries. Realistically the wordpress table names are never going to change, and the placeholders do obfuscate and complicate the code quite a lot. Though it's done like that throughout the codebase so no need to change it here if you want to stay consistent.

Reply:
I use wp var for table name because wordpress offer custom prefix option for table name at time of installation. if some external entity want to use our campaign feature they may not restrict with hardcoded table names. its an edge case but we cover it :)